scipy/integrate/quadpack.py

```diff
@@ -363,8 +363,40 @@ def quad(func, a, b, args=(), full_output=0, epsabs=1.49e-8, epsrel=1.49e-8,
         else:
             warnings.warn(msg, IntegrationWarning)
             return retval[:-1]
-    else:
-        raise ValueError(msg)
+
+    elif ier == 6:  # Forensic decision tree when QUADPACK throws ier=6
+        if epsabs <= 0:  # Small error tolerance - applies to all methods
+            if epsrel <= 50 * sys.float_info.epsilon or epsrel < 5e-29:
+                msg =- "If ``errabs<=0``, `epsrel` must be greater than both 5e-29 and 50*(machine epsilon)."
+            elif weight in ['sin', 'cos'] and (abs(a) + abs(b) == Inf):
+                msg = "sin- or cos- weighted intergals with infinite bounds must have ``epsabs>0``."
+
+        elif weight is None:
+            if points is None:  # QAGSE/QAGIE
+                msg = "Number of subintervals `limit` must be >=1."
+            else:  # QAGPE
+                if max(points) > max(a,b) or min(points) < min(a,b):
+                    msg = "All break points in `points` must lie within the integration bonds."
+                elif len(points) >= limit:
+                    msg = "Number of break points ({:d}) must be less than subinterval limit ({:d})".format(len(points), limit)
+
```

Comment:
PEP8 needs 80 char limit on each line

Reply:
Do you mean  for message output or source?  If the message:  it only takes 70.  Else: there are a lot of other (old) lines to change too.

Reply:
There are a lot of legacy non-PEP8-compliant files but since we are touching them we don't need to increase this noncompliancy I think. In the last few commits I've done the whole file even if I changed a tiny bit of it. There are also scripts for obvious PEP8 mistakes which you can diff to make sure bugs are not introduced.